packages/core/admin/admin/src/content-manager/hooks/useLazyComponents/index.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+import { useCustomFields } from '@strapi/helper-plugin';
+
+/**
+ * @description
+ * A hook to lazy load custom field components
+ * @param {Array.<string>} componentUids - The uids to look up components
+ * @returns object
+ */
+const useLazyComponents = (componentUids) => {
+  const [lazyComponentStore, setLazyComponentStore] = useState({});
+  const [loading, setLoading] = useState(true);
+  const customFieldsRegistry = useCustomFields();
+
+  useEffect(() => {
+    const lazyLoadComponents = async (uids, components) => {
+      const modules = await Promise.all(components);
+
+      uids.forEach((uid, index) => {
+        if (!Object.keys(lazyComponentStore).includes(uid)) {
+          setLazyComponentStore({ ...lazyComponentStore, [uid]: modules[index].default });
+        }
+      });
+    };
+
+    if (componentUids.length) {
+      const componentPromises = componentUids.map((uid) => {
+        const customField = customFieldsRegistry.get(uid);
```

Comment:
ℹ️  Instead of using this just for custom fields, in the future we could replace this with the [fields API](https://github.com/strapi/strapi/blob/main/packages/core/admin/admin/src/core/apis/Fields.js) if we ensure all fields added are dynamic imports and we add custom fields to the Fields object when the app starts.